kasa/tapoplug.py

```diff
@@ -0,0 +1,194 @@
+import base64
+import logging
+from datetime import datetime, timedelta
+from typing import Any, Dict, Optional, Set
+
+from .emeterstatus import EmeterStatus
+from .exceptions import AuthenticationException
+from .smartdevice import DeviceType, SmartDevice
+from .smartplug import SmartPlug
+from .tapoprotocol import TapoAESProtocol
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AttributeDict(dict):
+    __getattr__ = dict.__getitem__
+    __setattr__ = dict.__setitem__
+    __delattr__ = dict.__delitem__
+
+
+# TODO: there should be a baseclass for plugs that does not initialize modules etc. that are related only to some implementations
+class TapoPlug(SmartPlug):
+    def __init__(self, host: str, *, port: Optional[int] = None, credentials) -> None:
```

Comment:
```suggestion
    def __init__(self, host: str, *, port: Optional[int] = None, credentials: Optional[Credentials] = None, timeout: Optional[int]) -> None:
```

Oops, forgot to add this as I fixed it locally for testing.... This does not currently work as the function does not accept `timeout` kwarg as passed by `DeviceFactory`:
```
  File "/tmp/python-kasa/kasa/smartdevice.py", line 803, in connect
    return await connect(
           ^^^^^^^^^^^^^^
  File "/tmp/python-kasa/kasa/device_factory.py", line 64, in connect
    dev: SmartDevice = klass(
                       ^^^^^^
TypeError: TapoPlug.__init__() got an unexpected keyword argument 'timeout'

```

Reply:
Fixed